rust/chains/hyperlane-ethereum/src/interchain_gas.rs

```diff
@@ -136,7 +136,8 @@ where
         // TODO: if `SequenceIndexer` turns out to not depend on `Indexer` at all, then the supertrait
         // dependency could be removed, even if the builder would still need to return a type that is both
         // ``SequenceIndexer` and `Indexer`.
-        panic!("Gas payment nonce indexing not implemented");
+        info!("Gas payment nonce indexing not implemented");
```

Comment:
What is the case where we might hit this? It feels like returning 0 might lead to issues. Also might want a warn log instead of info if it indicates a misconfig.

Reply:
Random thought... Why not let the chain inform what indexing method to use? At least for now I suspect that every chain will only support one method, why have the config specify it at all.

Reply:
@tkporter actually had this idea but it seemed like a substantial task. As you suggested in the planning call today, I wanted to have the indexing just work at first, and only then do major refactoring but it could be a follow up task.